### PR TITLE
SeaMonkey Support

### PR DESCRIPTION
**Pull request by derinb**
SeaMonkey Support

### CODE REVIEW

File: extension/install.rdf [hunk 1]
```diff
@@ -61,6 +61,14 @@
       </Description>
     </em:targetApplication>
 
+    <em:targetApplication> <!-- SeaMonkey -->
+        <Description>
+            <em:id>{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}</em:id>
+            <em:minVersion>2.0</em:minVersion>
```
Comment (wagnerand) [inline extension/install.rdf]: Are you sure this is working in 2.0?

File: extension/content/customPersonaEditor.js [hunk 2]
```diff
@@ -263,6 +263,9 @@ var CustomPersonaEditor = {
       case PersonaService.FIREFOX_ID:
  	window.close();
         break;
+      case PersonaService.SEAMONKEY_ID:
+    window.close();
```
Comment (wagnerand) [inline extension/content/customPersonaEditor.js]: Bad formatting

File: extension/content/personas.css [hunk 3]
```diff
@@ -59,6 +59,11 @@
   height: auto !important;
 }
 
+#personas-toolbar-button {
```
Comment (wagnerand) [inline extension/content/personas.css]: Isn't this rule already defined in line 47?

File: extension/content/personas.css [hunk 4]
```diff
@@ -90,6 +95,20 @@
...
+#personas-toolbar-button.seamonkey, #personas-toolbar-button.seamonkey:hover, #personas-toolbar-button.seamonkey:hover:active {
+  list-style-image: url("chrome://personas/content/personas_30x30.png");
+  -moz-image-region: rect(0, 30px, 30px, 0px);  
+}
+
+toolbar[iconsize="small"] #personas-toolbar-button.seamonkey {
+  list-style-image: url("chrome://personas/content/personas_20x20.png");
+}
+
+#personas-menu.seamonkey {
```
Comment (wagnerand) [inline extension/content/personas.css]: Why do you need `!important` here? Can't this selector be just added to line 17?

File: extension/content/personas.css [hunk 5]
```diff
@@ -90,6 +95,20 @@
   font-weight: bold !important;
 }
 
+#personas-toolbar-button.seamonkey, #personas-toolbar-button.seamonkey:hover, #personas-toolbar-button.seamonkey:hover:active {
```
Comment (wagnerand) [inline extension/content/personas.css]: Are you sure the pseudo selectors are necessary?

File: extension/content/personas.css [hunk 6]
```diff
@@ -90,6 +95,20 @@
...
+  list-style-image: url("chrome://personas/content/personas_30x30.png");
+  -moz-image-region: rect(0, 30px, 30px, 0px);  
+}
+
+toolbar[iconsize="small"] #personas-toolbar-button.seamonkey {
+  list-style-image: url("chrome://personas/content/personas_20x20.png");
```
Comment (wagnerand) [inline extension/content/personas.css]: Is there a reason this does not have `-moz-image-region`?

File: extension/content/customPersonaEditor.js (hunk 2 above)
Comment (derinb) [inline extension/content/customPersonaEditor.js]: [Previos line](https://github.com/mozilla/personas-plus/pull/33#discussion-diff-70428007R264) is also bad formatted. Actually file already contains many formatting issues.

Though when I check the file with SublimeText the line 267 is aligned with line 264. I think Github is displaying file wrong here.

Let me prettify whole file after SeaMonkey PR.

Comment (wagnerand) [inline extension/content/customPersonaEditor.js]: I won't get into this discussion again. Please properly format new code you write. We can take care of the rest later.

File: extension/install.rdf (hunk 1 above)
Comment (derinb) [inline extension/install.rdf]: OK I will replace as. Tested on 2.26 and 2.45

```
			<em:minVersion>2.26</em:minVersion>
			<em:maxVersion>2.45</em:maxVersion>
```

File: extension/content/personas.css (hunk 5 above)
Comment (derinb) [inline extension/content/personas.css]: yes they are necessary. If not, other hover and active rules win and the toolbar icon is hidden.

File: extension/content/personas.css (hunk 6 above)
Comment (derinb) [inline extension/content/personas.css]: Actually I coded this 1 month ago, just waited the previous one to be approved. And I do not recall why I did not add the -moz-image-region.

But actually the personas_20x20.png image is already 20x20 so there is no need to place a  -moz-image-region property here.

Also because there is only 1 **toolbar[iconsize="small"]**  rule in this css file and this is specific to SeaMonkey Small Toolbar Icon state, the -moz-image-region is already in its inherited default value. So it seems there is no need for an extra -moz-image-region.

Comment (wagnerand) [inline extension/content/personas.css]: > But actually the personas_20x20.png image is already 20x20 so there is no need to place a -moz-image-region property here.

Isn't the same true for the 16*16 and 30*30 icon?
So, either declare `-moz-image-region` everywhere or nowhere, please.

Comment (derinb) [inline extension/content/personas.css]: But line 100 is required because **hover** and **active** pseudos change the -moz-image-region. To show the icon properly, -moz-image-region is set to 30px 30px as the image dimension.

File: extension/content/personas.css (hunk 4 above)
Comment (derinb) [inline extension/content/personas.css]: Line 47 belongs to Firefox toolbar button styling. Because it is 16x16px icon, I need to separate SeaMonkey toolbar button rules which require 20x20.

File: extension/content/personas.css (hunk 6 above)
Comment (wagnerand) [inline extension/content/personas.css]: Can you please declare it for this selector as well, then?

File: extension/content/personas.css (hunk 3 above)
Comment (derinb) [inline extension/content/personas.css]: Oh you are right, removed them.

File: extension/content/personas.css (hunk 4 above)
Comment (wagnerand) [inline extension/content/personas.css]: But this rule uses a 16x16 icon, the same as line 47.

File: extension/content/customPersonaEditor.js (hunk 2 above)
Comment (derinb) [inline extension/content/customPersonaEditor.js]: Andreas, if I prettify just the **3** lines I added, it does not align with above case and break statements. Because that **Switch** statement is already bad formatted itself.

I just tried to align with the already bad formatted lines.

If you want I can prettify all Switch statement but it will increase modified lines unnecessarily.

File: extension/content/personas.css (hunk 4 above)
Comment (derinb) [inline extension/content/personas.css]: Oh sorry this rule is for toolbar menu for SeaMonkey. I tried not to touch Firefox rules to separate SeaMonkey and Firefox rules.

I will remove !important.

File: extension/content/customPersonaEditor.js (hunk 2 above)
Comment (wagnerand) [inline extension/content/customPersonaEditor.js]: Ok, then please indent it at least correctly. line 267 having less indent than 266 just makes no sense.

File: extension/content/personas.css (hunk 6 above)
Comment (derinb) [inline extension/content/personas.css]: I think it is not necessary. Because there is no other rule for **toolbar[iconsize="small"]**, it does not get affected from hover or active pseudo rules. 

I tested on SeaMonkey and it works as expected with adding hover and active pseudos.

But let me know if you want to add them.

Comment (wagnerand) [inline extension/content/personas.css]: > I think it is not necessary.

Yeah I thought so. I still think it is better to add it for both safety and consistency reasons.

File: extension/content/customPersonaEditor.js (hunk 2 above)
Comment (derinb) [inline extension/content/customPersonaEditor.js]: OK indented by updated commit.

File: extension/content/personas.css (hunk 6 above)
Comment (derinb) [inline extension/content/personas.css]: Added by the new update

**Comment by derinb** [comment]
Andreas, do you see updated commit? Sometimes when I reload the page, it just disaapears from Github. There is an issue with Github I suppose.

File: extension/content/personas.css [hunk 7]
```diff
@@ -90,6 +90,20 @@
...
+#personas-toolbar-button.seamonkey, #personas-toolbar-button.seamonkey:hover, #personas-toolbar-button.seamonkey:hover:active {
+  list-style-image: url("chrome://personas/content/personas_30x30.png");
+  -moz-image-region: rect(0, 30px, 30px, 0px);  
+}
+
+toolbar[iconsize="small"] #personas-toolbar-button.seamonkey, toolbar[iconsize="small"] #personas-toolbar-button.seamonkey:hover, toolbar[iconsize="small"] #personas-toolbar-button.seamonkey:hover:active {
+  list-style-image: url("chrome://personas/content/personas_20x20.png");
+}
+
+#personas-menu.seamonkey {
```
Comment (wagnerand) [inline extension/content/personas.css]: Add this selector to the rule in line47.

File: extension/modules/personas.js [hunk 8]
```diff
@@ -134,7 +134,7 @@ var PersonaController = {
         }
     },
 
-    get_footer(document) {
+    get_footer: function(document) {
```
Comment (wagnerand) [inline extension/modules/personas.js]: All above changes are unrelated, please submit a new PR for them.

File: extension/content/personas.css (hunk 7 above)
Comment (derinb) [inline extension/content/personas.css]: Andreas, hover and active are already being handled by [other rules](personas.css#L65) for Firefox.

File: extension/modules/personas.js (hunk 8 above)
Comment (derinb) [inline extension/modules/personas.js]: This is actually added for SeaMonkey. It does not create any issue for Firefox. Without it SeaMonkey complains.

File: extension/content/personas.css (hunk 7 above)
Comment (wagnerand) [inline extension/content/personas.css]: Not sure what this has to do with `hover` or `active`. Those selectors are not used here.

Comment (derinb) [inline extension/content/personas.css]: Andreas, I did not understand what you actually ask for. Could you please more specific with lines and required changes?

Comment (wagnerand) [inline extension/content/personas.css]: So, the rules in 103 and 104 are the same as 48 and 49. So, instead of re-declaring them here, just add the selector `#personas-menu.seamonkey` to line 46/47 like this:
```
#personas-selector-button,
#personas-toolbar-button,
#personas-menu.seamonkey {
```

Comment (wagnerand) [inline extension/content/personas.css]: So, the rules in 103 and 104 are the same as 48 and 49. So, instead of re-declaring them here, just add the selector `#personas-menu.seamonkey` to line 46/47 like this:
```
#personas-selector-button,
#personas-toolbar-button,
#personas-menu.seamonkey {
```

Comment (derinb) [inline extension/content/personas.css]: Thanks Andreas for the details. Added.

File: extension/content/personas.css [hunk 9]
```diff
@@ -90,6 +91,15 @@
   font-weight: bold !important;
 }
 
+#personas-toolbar-button.seamonkey, #personas-toolbar-button.seamonkey:hover, #personas-toolbar-button.seamonkey:hover:active {
+  list-style-image: url("chrome://personas/content/personas_30x30.png");
+  -moz-image-region: rect(0, 30px, 30px, 0px);  
+}
+
+toolbar[iconsize="small"] #personas-toolbar-button.seamonkey, toolbar[iconsize="small"] #personas-toolbar-button.seamonkey:hover, toolbar[iconsize="small"] #personas-toolbar-button.seamonkey:hover:active {
```
Comment (wagnerand) [inline extension/content/personas.css]: Small nit-pick: Could you give each selector its own line (like lines 71-75 for example)? That would greatly enhance readability.

Comment (derinb) [inline extension/content/personas.css]: Done

**Comment by wagnerand** [comment]
Great, thank you!
